src/scandeval/text_to_text.py

```diff
@@ -0,0 +1,236 @@
+            model:
+                A pretrained model. Can be None if the model is not used in the
+                initialisation of the data collator. Defaults to None.
+
+        Returns:
+            The data collator.
+        """
+        return DataCollatorWithPadding(tokenizer, padding="longest")
+
+    def _compute_metrics(
+        self,
+        model_outputs_and_labels: tuple[np.ndarray, np.ndarray],
+        id2label: list[str],
+    ) -> dict[str, float]:
+        """Compute the metrics needed for evaluation.
+
+        Args:
+            model_outputs_and_labels:
+                The first sequence contains the model outputs and the second sequence
+                contains the true labels.
+            id2label (list of str):
+                Conversion of indices to labels.
+
+        Returns:
+            A dictionary with the names of the metrics as keys and the metric values as
+            values.
+        """
+        model_outputs, labels = model_outputs_and_labels
+
+        model_output_dtype = np.asarray(model_outputs).dtype
+        if model_output_dtype in [np.float16, np.float32, np.float64]:
+            predictions = np.asarray(model_outputs).argmax(axis=-1)
+        else:
+            predictions = model_outputs
+
+        results: dict[str, float] = dict()
+        for cfg in self.dataset_config.task.metrics:
+            metric = self._metrics[cfg.name]
+            score_dict: dict[str, float] | None = metric.compute(
+                predictions=predictions,
+                references=labels,
+                **cfg.compute_kwargs,
+            )
+            if score_dict is not None:
+                scores = score_dict[cfg.results_key]
+                if isinstance(scores, list):
+                    scores = sum(scores) / len(scores)
+                results[cfg.name] = scores
+        return results
+
+    def _extract_few_shot_examples(
+        self, train_dataset: Dataset, random_seed: int
+    ) -> list[dict[str, Any]]:
+        """Extract few-shot examples from the training dataset.
+
+        Args:
+            train_dataset:
+                The training dataset.
+            random_seed:
+                The random seed to use when extracting the few-shot examples.
+
+        Returns:
+            The few-shot examples.
+        """
+        shuffled_train = train_dataset.shuffle(seed=random_seed)
+        num_few_shots = self.dataset_config.num_few_shot_examples
+        few_shot_examples: list[dict[str, Any]] = list()
+        while len(few_shot_examples) < num_few_shots:
+            example = shuffled_train.select(range(1))[0]
+            few_shot_examples.append(example)
+            shuffled_train = shuffled_train.filter(
+                lambda x: x["text"] != example["text"]
+            )
+        random.seed(random_seed)
+        random.shuffle(few_shot_examples)
+        return few_shot_examples
+
+    def _apply_few_shot_prompt(
```

Comment:
You mean the prompt template is model-dependent? Because this function simply concatenates the few-shot examples with the final example. The prompt templates are all located in the `dataset_configs` module.